internal/scaffold/ansible/watches.go

```diff
@@ -39,6 +53,57 @@ func (w *Watches) GetInput() (input.Input, error) {
 	return w.Input, nil
 }
 
+// UpdateAnsibleWatchForResource checks for duplicate GVK, and appends to existing Watch.yaml file.
+func UpdateAnsibleWatchForResource(r *scaffold.Resource, absProjectPath string) error {
+	watchFilePath := filepath.Join(absProjectPath, WatchesFile)
+	watchYAML, err := ioutil.ReadFile(watchFilePath)
+	if err != nil {
+		log.Error(err, "failed to read watch manifest %v")
+		return nil
+	}
+
+	watchList := []watchesYAML{}
+	err = yaml.Unmarshal(watchYAML, &watchList)
+	if err != nil {
+		log.Error(err, "Failed to unmarshal watch config")
+		return nil
+	}
+	gvk := schema.GroupVersionKind{
+		Version: r.Version,
+		Group:   r.FullGroup,
+		Kind:    r.Kind,
+	}
+	for _, watch := range watchList {
+		if watch.GroupVersionKind == gvk {
+			// dupe detected
+			return fmt.Errorf("duplicate GVK: %v", watch.GroupVersionKind.String())
+		}
+	}
+
+	// Create new watch
+	newWatch := watchesYAML{
+		GroupVersionKind: gvk,
+		Role:             r.LowerKind,
+	}
+
+	// Append new Watch content existing watch.yaml
+	watchList = append(watchList, newWatch)
+
+	// Marshal New WatchYAML File
+	data, err := yaml.Marshal(watchList)
+	if err != nil {
+		log.Error(err, "Failed to Marshal watch config")
+		return nil
+	}
+
+	if err := ioutil.WriteFile(watchFilePath, data, fileutil.DefaultFileMode); err != nil {
+		return fmt.Errorf("failed to update %v: %v", watchFilePath, err)
+	}
+
+	return nil
+}
+
+// template to create new Watch.yaml content.
```

Comment:
For follow-up: with this PR, we'll be using a string template for creating and a struct/YAML marshaling for updating. We should consolidate to using the struct/YAML marshaling method for both.

Can you add a TODO comment for this?